src/index.ts

```diff
@@ -252,13 +264,18 @@ export class VuexPersistence<S> implements PersistOptions<S> {
         }
 
         this.subscriber(store)((mutation: MutationPayload, state: S) => {
-          if (this.filter(mutation)) {
+          if (!this.shuttingDown && this.filter(mutation)) {
             this.saveState(this.key, this.reducer(state), this.storage)
           }
         })
 
         this.subscribed = true
       }
+
+      this.prepareShutdown = () => {
+        this.shuttingDown = true;
```

Comment:
This will just prevent future writes from happening and doesn't do any waiting. Is that the goal?